### PR TITLE
Traffic Router DNS zone calculation optimizations

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR is not related to any issue. It optimizes the way Traffic Router (TR) calculates the zone(delivery service) of an incoming reuest.
<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Router

## What is the best way to verify this PR?

Make sure all the tests pass.
Run TR locally and make DNS/ HTTP requests against it. If the request is of the form `routing name.DS FQDN`, the resultant delivery service should be returned from the new map that was introduced in this PR.

## If this is a bugfix, which Traffic Control versions contained the bug?

- master

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/edge/CacheRegister.java [hunk 1]
```diff
@@ -140,14 +143,42 @@ public void setDeliveryServiceMatchers(final TreeSet<DeliveryServiceMatcher> mat
 		this.deliveryServiceMatchers = matchers;
 	}
 
+	private String getRequestName(final Request request) {
```
Comment (rawlinp) [inline traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/edge/CacheRegister.java:146]: `Request` has a `getHostname()` method -- does that not work here? It appears to be set for both DNS and HTTP requests, but I imagine for DNS there might be a trailing `.` to strip off before doing a map lookup.

Comment (srijeet0406) [reply traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/edge/CacheRegister.java:146]: You're right, I could totally use the `getHostname()` method. Also, looks like it already strips out the trailing `.` for DNS requests.

**Review by rawlinp — APPROVED** [review]

**Comment by zrhoffman** [comment]
> The TR Ultimate Test Harness is failing for an unrelated reason, it seems to always fail on startup, for any of a number of reasons.

Fixed in #7637

**Comment by zrhoffman** [comment]
Reverted #7622 in c6fcfea717 due to DNS load tests consistently failing with #7622 but passing with it reverted.

@srijeet0406 Would you please re-PR #7622 now that we have performance test coverage again?

**Pull request merged**